### ISSUE TITLE
Expecting the tensors input of MTCNN

### ISSUE THREAD

**Issue by sweihub** [timesler/facenet-pytorch#115]
Hi

Current MTCNN accepts PIL Image as input, I am seeking a MTCNN accepts tensors as input which dose not break the pytorch gradients, by doing this, we can build a face similarity loss function. Is this possible?

My expecting usage
```
def face_similarity(tensor1, tensor2):
    face1 = mtcnn(tensor1)
    face2 = mtcnn(tensor2)
    embedding1 = ResNet(face1.unsqueeze(0))
    embedding2 = ResNet(face2.unsqueeze(0))
    return torch.cdist(embedding1, embedding2)


# Training
loss = 1 - face_similarity(tensor1, tensor2)
loss.backward()

```

Thanks

**Comment by timesler** [comment]
@sweihub you should not need to train the MTCNN model to implement this, and I'm not sure it makes sense to do so. The face similarity metrics you are trying to implement represents a distance between two face vectors, and is distinct from the face detection task. 

Why not use the following?

```python
def face_similarity(tensor1, tensor2):
    with torch.no_grad():
        face1 = mtcnn(tensor1)
        face2 = mtcnn(tensor2)
    embedding1 = ResNet(face1.unsqueeze(0))
    embedding2 = ResNet(face2.unsqueeze(0))
    return torch.cdist(embedding1, embedding2)
```
